documents/src/pages/elements/text-field.md

```diff
@@ -48,9 +48,118 @@ p {
 
 ::
 
-`ef-text-field` is a form element for text.
+`ef-text-field` is a form element for text. This basic text input enables the user to type a small amount of text and submit it into Workspace.
 
-## Usage
+## Design Usage Rules
+
+### Content principles
+
+* Users should understand the state of the field quickly and easily
+* Placeholder text should not contain instructions. If these are needed supportive text should be provided
+* Label text can wrap onto multiple lines if needed. Avoid truncating, where possible  
+* Supportive  text should be used only where needed
+
+### Formatting
+
+Form labels should be visible and there are 3 variants:
+
+#### Stacked
+
+Stacked is the default form label location
+
+::
+
+```javascript
+::text-field::
+```
+
+```css
+label {
+  display: block;
+}
+ef-text-field {
+  width: 240px;
+}
+```
+
+```html
+<label id="name">Full Name</label>
+<ef-text-field 
+  aria-labelledby="name"
+  placeholder="Your name as shown on your passport">
+</ef-text-field>
+```
+
+::
+
+#### Left Label
+
+Use a left form label when space is vertically limited
+
+::
+
+```javascript
+::text-field::
+```
+
+```css
+ef-text-field {
+  width: 240px;
+}
+```
+
+```html
+<label id="name">Full Name</label>
+<ef-text-field 
+  aria-labelledby="name"
+  placeholder="Your name as shown on your passport">
+</ef-text-field>
+```
+
+::
+
+#### Hidden Label
+
+The form label can be hidden when needed as long as an aria-label is included
+
+::
+
+```javascript
+::text-field::
+```
+
+```css
+ef-text-field {
+  width: 240px;
+}
+```
+
+```html
+<ef-text-field 
+  aria-label="Full name"
+  placeholder="Your name as shown on your passport">
+</ef-text-field>
+```
+
+::
+
+### Copy and UX Writing
+
+See the <a href="https://lsegroup.sharepoint.com/sites/ProductDesignTeam/SitePages/Content-Design.aspx" target="_blank" rel="noreferrer">Content Design Style Guide (internal only)</a>
+
+### Touch Devices and Virtual Keyboards
+
+When interacting with `ef-text-field` using a mobile or touch device ensure that the correct virtual keyboard is activated upon user focus.
+
+### Validation and Errors
+
+Use inline validation where possible. For example, if the user typed an email address into a name field. Show the user an error message when we know the user has made an error. If it is not possible to use inline validation, then validate when the user submits the entry. For example, on a form.
+
+Inline validation is when validation messages are shown immediately after the user types in data to form fields.
```

Comment:
@Dave-Musson  @TremayneChrist Does this mean validating the input value while user is typing? We experience that this pattern causes some issues and a bit tricky to implement it. Mainly, it's because it's hard to know if user has done typing or has not done the typing. IMO, validate the value after focussed out would simplify this and more consistency way?

Reply:
We questioned this with Ole as its content coming from Rise. Ole confined, this is how EF works, so it was kept. 
Check with Derek and Matt D

Reply:
let me clarify that immediate validation is possible but it doesn't mean we should do it. It has various issues as described in https://adamsilver.io/blog/live-validation-is-problematic/.

Reply:
For sure, it is - zero dispute. There a better methods
However, it's what the code does, seeing the examples in the implementation section.
The behaviour was kept with the phase 0 scope to **avoid code changes.**
If we want to change this (which I'd recommend), why aren't we also addressing the other rUX and accessibility defects in the component?

Reply:
Correct = Phase 0 isn't designed to improve the experience, its to mimic in Figma what's in EF and the current functionality / behaviour.

Reply:
I have clarified this to Matt in the meeting. It has more detail around validation. If developers use `maxLength` `minLength` or `pattern` attributes, Text Field will do inline validation -- this is built-in feature in Text Field and it's similar to native input. However, if they don't use those attributes, Text Field won't do any validation and delegate the logic back to developers. Developer can then use `error` attribute to show read border. 

So in term of custom validation, both inline validation or validation when focussed-out are doable, this is up to developer and specification.

If the statement in the doc is about built-in feature in Text Field, that's 100% correct. If the statement is meant to tell  users what they should do when they do custom validation, it's still correct :) It depends on what pattern that we to advertise.

Reply:
We still need to chat this through to ensure we're on the same page and not creating functionality which didn't currently exist. OR if no OOTB validation behaviour currently exists what are we doing. I don't beleive it should be about what we CAN do, its whats closest to existing behaviour. @Dave-Musson @dtanp-rft  lets just have a chat

Reply:
Let's keep the copy as it is. Resolving the conversion.